compare/README.md

```diff
@@ -65,6 +66,7 @@ Only 3 of 4 images have passed test
 For each target image file the program reports a maximal ratio of the per pixel comparison error to the baseline error scale.
 The baseline error scale tries to take account of neighboring pixels to estimate visibility of an error in that specific pixel location.
 Currently the range of [0–2.0] is marked as `OK`, the range of [2.0–4.0] as `BAD` and more than 4.0 as `FAIL`.
+In the bitwise comparison mode (`-s 0`) any error value larger than zero is marked as `FAIL`.
```

Comment:
It would be more useful, to preserve the categorisation and let compare return a different value depending on the maximum deviation level in strict/detailed mode. Eg: `0` for flawless, `1` for `OK` (but not all zero), `2` for `BAD`, `3` for `FAIL` and some other larger value for program errors.
This conceptually allows to eg mark CI runs as `neutral` when the result isn't flawless but not yet bad enough for a `FAIL`, once either GHA re-adds neutral success states or we switch the CI provider again.

Reply:
Should I add additional flag for that: `-p <pass-level:0-3>`? Each image gets its own result flag (0:SAME, 1:OK, 2:BAD, 3:FAIL, 4:ERROR) and if that flag is less or equal than pass level then image passes. If all images pass then `compare` returns 0, otherwise returns maximum result flag. So default pass level of 2 is equivalent to the current behavior and pass level 0 is bitwise comparison mode. Also pass level 3 may be useful: it's error checking mode that doesn't care about image comparison results at all.

Reply:
Won't just having different return codes allow all the same functionality as `-p <pass-level>` while being more flexible?

Reply:
I. e. always have `-p 0`? I can see two problems:
* Program returns nonzero even if all images pass. Not sure how problematic it can be.
* Old format for final summary isn't enough. Well, I can do something about it, for example print counts for all categories.

Reply:
> I. e. always have -p 0?

No, those return values would only be used if a flag for strict/detailed mode was passed, eg plain `-c` without any arguments  *(it's strict in the sense that anything but flawless returns non-zero, and detailed because the return code specifies how much the worst results deviated)*. If I understood you correctly, the `-p <pass-level>` you proposed only ever returns binary “worst result was below/above specified level”.
With this binary¹ result it would be impossible to classify “flawless” as OK, “not failed” as neutral and “FAIL” as failed in the CI — unless you run the tests twice. Even if the result is non-zero, it is still possible to use this strict mode with `set -e`-scripts, by eg using something like this:
```sh
#/bin/sh
set -e
ret=0
compare . -c $COMP_ARGS || ret="$?"
if [ "$ret" -eq 0 ] ; then
  echo "::set-result=OK"
elif [ "$ret" -le "2" ] ; then
  echo "::set-result=NEUTRAL"
else
  echo "::set-result=FAIL"
fi
```

The `-p <pass-level>` option could be emulated with a simple `test "$ret" -le "$pass_level"`.
```

---

¹: ternary with program error

Reply:
> No, those return values would only be used if a flag for strict/detailed mode was passed, eg plain `-c` without any arguments

No, I mean to replace `-c` with `-p`. So instead of a simple binary flag I'm planning to use more flexible approach.

> If I understood you correctly, the `-p <pass-level>` you proposed only ever returns binary “worst result was below/above specified level”.

I'm planning to return `(worst <= pass_level ? 0 : worst)`, so `-p 0` would be exactly equivalent to `-c`.

Reply:
> I'm planning to return (worst <= pass_level ? 0 : worst), so -p 0 would be exactly equivalent to -c.

Well, then at least for the return code `-p <pass-level>` is fine and basically provides a convenient  "automatic-`(test "$ret" -le "$pass" || exit "$ret")`".

But still a nit: marking everything as `FAIL` in the log mkaes it not impossible, but a tad harder to tell which "failing" tests failed how bad, as you actually need to compare all numbers and lose the classification into groups for everything above `pass-level`. But as I wrote, that's more a nit.

Reply:
I'm planning to keep uniform log independent of `-p` level. Only final summary and exit code would be affected.

Reply:
In that case, the option is fine, but the Documentation in README should state that the return values differ and not just state “error value larger than zero is marked as `FAIL`.”.

Reply:
Done.